test/preview/components/embedders/test_openai_document_embedder.py

```diff
@@ -0,0 +1,329 @@
+                    "prefix ML | A transformer is a deep learning architecture suffix",
+                ],
+            )
+        documents_with_embeddings = result["documents"]
+        metadata = result["metadata"]
+
+        assert isinstance(documents_with_embeddings, list)
+        assert len(documents_with_embeddings) == len(docs)
+        for doc in documents_with_embeddings:
+            assert isinstance(doc, Document)
+            assert isinstance(doc.embedding, list)
+            assert len(doc.embedding) == 1536
+            assert all(isinstance(x, float) for x in doc.embedding)
+        assert metadata == {"model": model, "usage": {"prompt_tokens": 4, "total_tokens": 4}}
+
+    @pytest.mark.unit
+    def test_run_custom_batch_size(self):
+        docs = [
+            Document(text="I love cheese", metadata={"topic": "Cuisine"}),
+            Document(text="A transformer is a deep learning architecture", metadata={"topic": "ML"}),
+        ]
+
+        model = "text-similarity-ada-001"
+        with patch(
+            "haystack.preview.components.embedders.openai_document_embedder.openai.Embedding"
+        ) as openai_embedding_patch:
+            openai_embedding_patch.create.side_effect = mock_openai_response
+            embedder = OpenAIDocumentEmbedder(
+                api_key="fake-api-key",
+                model_name=model,
+                prefix="prefix ",
+                suffix=" suffix",
+                metadata_fields_to_embed=["topic"],
+                embedding_separator=" | ",
+                batch_size=1,
+            )
+
+            result = embedder.run(documents=docs)
+
+            assert openai_embedding_patch.create.call_count == 2
+
+        documents_with_embeddings = result["documents"]
+        metadata = result["metadata"]
+
+        assert isinstance(documents_with_embeddings, list)
+        assert len(documents_with_embeddings) == len(docs)
+        for doc in documents_with_embeddings:
+            assert isinstance(doc, Document)
+            assert isinstance(doc.embedding, list)
+            assert len(doc.embedding) == 1536
+            assert all(isinstance(x, float) for x in doc.embedding)
+
+        # openai.Embedding.create is called 2 times
+        assert metadata == {"model": model, "usage": {"prompt_tokens": 2 * 4, "total_tokens": 2 * 4}}
+
+    @pytest.mark.unit
+    def test_run_wrong_input_format(self):
+        embedder = OpenAIDocumentEmbedder(api_key="fake-api-key")
+
+        # wrong formats
+        string_input = "text"
+        list_integers_input = [1, 2, 3]
+
+        with pytest.raises(TypeError, match="OpenAIDocumentEmbedder expects a list of Documents as input"):
+            embedder.run(documents=string_input)
+
+        with pytest.raises(TypeError, match="OpenAIDocumentEmbedder expects a list of Documents as input"):
+            embedder.run(documents=list_integers_input)
+
+        # empty list is ok
```

Comment:
I would not expect this in the `wrong_input_format` test. Let's make it a separate unit test. It can easily happen that a retriever returns an empty list and passes it on to a retriever for example. I wouldn't call it a wrong input format.